r4/src/main/java/org/opencds/cqf/r4/providers/DataRequirementsProvider.java

```diff
@@ -96,6 +223,23 @@ public CqfMeasure createCqfMeasure(Measure measure,
         return this.createCqfMeasure(measure, libraryMap);
     }
 
+    public org.hl7.fhir.r4.model.Library getLibraryFromMeasure(Measure measure,
+               LibraryResolutionProvider<org.hl7.fhir.r4.model.Library> libraryResourceProvider) {
+
+        org.hl7.fhir.r4.model.Library primaryLibrary = null;
+        Iterator var6 = measure.getLibrary().iterator();
+
+        String library= null;
+        while(var6.hasNext()) {
+            org.hl7.fhir.r4.model.CanonicalType ref = (org.hl7.fhir.r4.model.CanonicalType) var6.next();
+            library = (String) ref.getValue();
```

Comment:
This should use the first library if there are multiple. There are some additional conformance requirements that constrain this to one Library.